### PR TITLE
fix: Use name of other participant on archived rooms

### PR DESCRIPTION
**Pull request by Pat992**
Previously archived rooms where called Empty chat (was <Your own username>).
Now the other username is selected in case of direct chats that are archived, via the room-member-states and client-userID

### CODE REVIEW

File: lib/src/room.dart [hunk 1]
```diff
@@ -290,6 +290,19 @@ class Room {
...
     }
+    if (isArchived) {
+      // In case of a direct chat, we need to get the room-members via the RoomMember event-type
+      // then we have to make sure to get the other room-member that is not us, to add them to the display name.
+      final roomMemberEvents = states.entries
+          .firstWhereOrNull((element) => element.key == EventTypes.RoomMember);
+      final otherRoomMember = roomMemberEvents?.value.entries
+          .firstWhereOrNull((entry) => entry.value.senderId != client.userID);
+
+      if (otherRoomMember?.value.prevContent?.tryGet('displayname') != null) {
```
Comment (td-famedly) [inline lib/src/room.dart:301]: why prevContent here?

File: lib/src/room.dart [hunk 2]
```diff
@@ -290,6 +290,19 @@ class Room {
           .calcDisplayname(i18n: i18n);
       if (sender != null) return sender;
     }
+    if (isArchived) {
```
Comment (td-famedly) [inline lib/src/room.dart:293]: Could just go in the membership.leave block

File: lib/src/room.dart [hunk 3]
```diff
@@ -290,6 +290,19 @@ class Room {
...
+      // In case of a direct chat, we need to get the room-members via the RoomMember event-type
+      // then we have to make sure to get the other room-member that is not us, to add them to the display name.
+      final roomMemberEvents = states.entries
```
Comment (td-famedly) [inline lib/src/room.dart:296]: hm idm this, but you could also just use directChatMatrixId to simplify this

File: lib/src/room.dart [hunk 4]
```diff
@@ -291,6 +291,11 @@ class Room {
       if (sender != null) return sender;
     }
     if (membership == Membership.leave) {
+      if (directChatMatrixID != null) {
```
Comment (td-famedly) [inline lib/src/room.dart:294]: I would merge this with the lines just below this, looks like prev_sender is never sent in member events

**Review by td-famedly — CHANGES_REQUESTED** [review]

**Review by krille-chan — APPROVED** [review]
LGTM 👍

**Pull request merged**